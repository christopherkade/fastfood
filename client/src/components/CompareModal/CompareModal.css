details {
  padding: 1rem 0;
}
details summary {
  display: inline-block;
  cursor: pointer;
  padding: 8px 12px;
  padding-top: 10px;
  border-radius: 4px;
  background: #00749a;
  color: #fff;
  font-weight: bold;
  text-transform: uppercase;
}

details summary:-webkit-details-marker {
  display: none;
}

details[open] summary {
  background: #EF798A;
}

details[open] summary:before {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: block;
  cursor: default;
  content: '';
  z-index: 99;
  background: rgba(27, 31, 35, 0.5);
}

details-dialog {
  left: 50%;
  margin: 10vh auto;
  max-height: 80vh;
  max-width: 90vw;
  position: fixed;
  top: 0;
  transform: translateX(-50%);
  width: 80%;
  z-index: 999;
  animation-name: fadein;
  animation-duration: 300ms;
  animation-timing-function: ease-in-out;
  background-color: #fff;
  border: 1px solid #EF798A;
  border-top: 10px solid #EF798A;
  border-radius: 3px;
  padding: 1em;
  box-shadow: 1px 1px 10px rgba(0, 0, 0, 0.3);
}

@keyframes fadein {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
