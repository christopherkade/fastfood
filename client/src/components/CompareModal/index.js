export { default as CompareModal } from "./CompareModal"
