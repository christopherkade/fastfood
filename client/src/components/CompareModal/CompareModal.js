import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

import "./CompareModal.css"

const Title = styled.h1`
  font-size: 2rem;
`

const StyledList = styled.ul`
  list-style: none;
`

/**
 * Sort and clean the food array
 * @param {*} foodCount
 */
const sortFood = (foodCount) => {
  const keys = Object.keys(foodCount)
  const trimmedFood = keys.map(key => {
    if (key === "") return null

    const { name, emoji } = foodCount[key].obj

    return {
      name,
      emoji,
      count: foodCount[key].count
    }
  })
    .filter(food => food)
    .sort((a, b) => a.count - b.count)
    .reverse()

  return trimmedFood
}

const CompareLayout = ({ count, onClick }) => {
  let trimmedFood = sortFood(count)

  return (
    <details-dialog>
      <Title>Vote count</Title>
      <StyledList>
        {trimmedFood.map(food => <li key={food.name}>x{food.count} {food.emoji} ({food.name}) </li>)}
      </StyledList>
      <button onClick={onClick}>Close</button>
    </details-dialog>
  )
}

CompareLayout.propTypes = {
  count: PropTypes.object
}

CompareLayout.defaultProps = {
  count: {}
}

export default CompareLayout
