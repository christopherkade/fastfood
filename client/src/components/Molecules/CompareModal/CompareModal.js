import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { HorizontalBar } from "react-chartjs-2"

import "./CompareModal.css"
import { Button } from "components/Atoms/Button"
import { Title } from "components/Atoms/Title"

const Subtitle = styled.h2`
  font-size: 1.5rem;
  font-weight: 300;
`

/**
 * Sort and clean the food array
 * @param {*} foodCount
 */
const extractDataFromFood = (foodCount) => {
  const labels = []

  const datasets = [
    {
      label: 'Food choices',
      backgroundColor: 'rgba(255,99,132,0.2)',
      borderColor: 'rgba(255,99,132,1)',
      borderWidth: 1,
      hoverBackgroundColor: 'rgba(255,99,132,0.4)',
      hoverBorderColor: 'rgba(255,99,132,1)',
      data: []
    }
  ]

  const keys = Object.keys(foodCount)

  keys.map(key => {
    if (key === "") return null

    const { name, emoji } = foodCount[key].obj
    return {
      name,
      emoji,
      count: foodCount[key].count
    }
  })
    .filter(food => food)
    .sort((a, b) => a.count - b.count)
    .reverse()
    .forEach(({ name, emoji, count }) => {
      labels.push(emoji)
      datasets[0].data.push(count)
    })

  return {
    labels,
    datasets
  }
}

const CompareModal = ({ count, onClick }) => {
  let foodData = extractDataFromFood(count)

  const options = {
    legend: {
      display: false
    },
    scales: {
      xAxes: [{
        ticks: {
          beginAtZero: true,
        }
      }],
      yAxes: [{
        ticks: {
          fontSize: 30
        }
      }]
    }
  }

  return (
    <details-dialog>
      <Title>Vote count</Title>
      {
        foodData.labels.length > 0 ?
          <HorizontalBar
            data={foodData}
            options={options}
          /> :
          <Subtitle>
            It seems like nobody has picked any food yet ! <span role="img" aria-label="">ü§∑‚Äç‚ôÇ</span>Ô∏è
          </Subtitle>
      }
      <Button onClick={onClick}>Close</Button>
    </details-dialog>
  )
}

CompareModal.propTypes = {
  count: PropTypes.object,
  onClick: PropTypes.func.isRequired,
}

CompareModal.defaultProps = {
  count: {},
}

export default CompareModal
