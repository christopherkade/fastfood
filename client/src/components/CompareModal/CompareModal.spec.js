import React from 'react';
import CompareModal from './CompareModal';
import { render, cleanup } from 'react-testing-library';

afterEach(cleanup)

jest.mock("react-chartjs-2", () => ({
  HorizontalBar: "canvas"
}))

describe('<CompareModal />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<CompareModal onClick={() => { }} />);
      expect(container).toMatchSnapshot();
    });
  });
});
