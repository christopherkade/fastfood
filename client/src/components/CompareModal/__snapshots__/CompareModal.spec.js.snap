// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompareModal /> Snapshots should match snapshot 1`] = `
<div>
  <details-dialog>
    <h1
      class="sc-bwzfXH gllGzP"
    >
      Vote count
    </h1>
    <canvas
      data="[object Object]"
      options="[object Object]"
    />
    <button
      class="sc-bdVaJa cnrKsD"
    >
      Close
    </button>
  </details-dialog>
</div>
`;
